src/nodes.coffee

```diff
@@ -1489,21 +1489,32 @@ exports.Obj = class Obj extends Base
     return yes for prop in @properties when prop instanceof Splat
     no
 
+  # Move rest property to the end of the list.
+  # `{a, rest..., b} = obj` -> `{a, b, rest...} = obj`
+  # `foo = ({a, rest..., b}) ->` -> `foo = {a, b, rest...}) ->`
+  reorderProperties: ->
+    props = @properties
+    splatProps = (prop for prop in props when prop instanceof Splat)
+    objProps = (prop for prop in props when prop not instanceof Splat)
+    @objects = @properties = [].concat objProps, splatProps
+
   compileNode: (o) ->
+    @reorderProperties() if @hasSplat() and @lhs
     props = @properties
     if @generated
       for node in props when node instanceof Value
         node.error 'cannot have an implicit value in an implicit object'
 
-    # Object spread properties. https://github.com/tc39/proposal-object-rest-spread/blob/master/Spread.md
-    return @compileSpread o if @hasSplat() and not @csx
-
     idt      = o.indent += TAB
     lastNode = @lastNode @properties
 
     # CSX attributes <div id="val" attr={aaa} {props...} />
     return @compileCSXAttributes o if @csx
 
+    if @hasSplat() and @lhs
```

Comment:
why not do this check in `@reorderProperties`?

Reply:
Nice catch. Thanks.